hibernate-core/src/test/java/org/hibernate/orm/test/loading/multiLoad/MultiNaturalIdLoadTest.java

```diff
@@ -0,0 +1,433 @@
+				session -> {
+					session.setCacheMode( CacheMode.IGNORE );
+					for ( int i = 1; i <= 10; i++ ) {
+						session.persist( new SimpleNaturalIdEntity( i, "Entity" + i ) );
+					}
+					for ( int i = 1; i <= 10; i++ ) {
+						session.persist( new CompositeNaturalIdEntity( i, "Entity" + i, i + "Entity" ) );
+					}
+				}
+		);
+	}
+
+	@AfterEach
+	public void tearDown(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					session.createMutationQuery( "delete SimpleNaturalIdEntity" ).executeUpdate();
+					session.createMutationQuery( "delete CompositeNaturalIdEntity" ).executeUpdate();
+				}
+		);
+	}
+
+	@Test
+	public void testBasicUnorderedMultiLoad(SessionFactoryScope scope) {
+		final SQLStatementInspector statementInspector = scope.getCollectingStatementInspector();
+		scope.inTransaction(
+				session -> {
+					statementInspector.getSqlQueries().clear();
+
+					List<SimpleNaturalIdEntity> results = session
+							.byMultipleNaturalId( SimpleNaturalIdEntity.class )
+							.enableOrderedReturn( false )
+							.multiLoad( "Entity1","Entity2","Entity3","Entity4","Entity5" );
+					assertEquals( 5, results.size() );
+
+					Iterator<SimpleNaturalIdEntity> it = results.iterator();
+					for ( int i = 1; i <= 5; i++ ) {
+						SimpleNaturalIdEntity se= it.next();
+						if ( i == se.getId() ) {
+							it.remove();
+						}
+					}
+					assertEquals( 0, results.size() );
+
+					final int paramCount = StringHelper.countUnquoted(
+							statementInspector.getSqlQueries().get( 0 ),
+							'?'
+					);
+
+					final Dialect dialect = session.getSessionFactory()
+							.getJdbcServices()
+							.getDialect();
+					if ( MultiKeyLoadHelper.supportsSqlArrayType( dialect ) ) {
+						assertEquals(1, paramCount );
+					}
+					else {
+						assertEquals(5, paramCount );
+					}
+				}
+		);
+	}
+
+	@Test
+	public void testBasicOrderedMultiLoad(SessionFactoryScope scope) {
+		final SQLStatementInspector statementInspector = scope.getCollectingStatementInspector();
+		scope.inTransaction(
+				session -> {
+					statementInspector.getSqlQueries().clear();
+
+					List<String> ids = List.of("Entity4","Entity2","Entity5","Entity1","Entity3");
+					List<SimpleNaturalIdEntity> results = session
+							.byMultipleNaturalId( SimpleNaturalIdEntity.class )
+							.enableOrderedReturn( true )
+							.multiLoad( ids );
+
+					assertEquals( 5, results.size() );
+					for ( int i = 0; i < 5; i++ ) {
+						assertEquals(ids.get(i), results.get(i).getSsn() );
+					}
+
+					final int paramCount = StringHelper.countUnquoted(
+							statementInspector.getSqlQueries().get( 0 ),
+							'?'
+					);
+
+					final Dialect dialect = session.getSessionFactory()
+							.getJdbcServices()
+							.getDialect();
+					if ( MultiKeyLoadHelper.supportsSqlArrayType( dialect ) ) {
+						assertEquals(1, paramCount );
+					}
+					else {
+						assertEquals(5, paramCount );
+					}
+				}
+		);
+	}
+
+	@Test
+	@SkipForDialectGroup(
```

Comment:
This skip blocks here and below are a bit confusing IMO as there is no javadoc comment on why the skips are necessary. Also, I'm a bit surprised that a multi-load of 3 composite natural ids only requires 3 parameters. What is rendered SQL for this? I would expect this always requires 6 parameters i.e. `where (ssn,ssn2) in ( (?,?), (?,?), (?,?) )` because tuples can't be passed as a single parameter AFAIK.

Reply:
The regex that i'm using for the tests that exclude Sybase, SQLServer, DB2 and hsqldb are looking at the (?,?) pattern, ergo, it checks whether there are 3
Sybase, SQLServer, DB2 and hsqldb produce different sql queries that effectively contain 6 '?', but using `MultiKeyLoadHelper.supportsSqlArrayType` for this didn't do the trick
I'll add a comment to the tests to clarify this.